src/generator/AutoRest.Ruby.Azure/Model/CompositeTypeRba.cs

```diff
@@ -42,15 +50,49 @@ public override string GetBaseTypeName()
                 if (this.BaseModelType.Extensions.ContainsKey(AzureExtensions.ExternalExtension) ||
                     this.BaseModelType.Extensions.ContainsKey(AzureExtensions.AzureResourceExtension))
                 {
-                    typeName = "MsRestAzure::" + typeName;
+                    if(!nameRegEx.IsMatch(typeName) || !IsResourceModelsMatchStandardDefinition(this))
+                    {
+                        typeName = "MsRestAzure::" + typeName;
+                    }
                 }
 
                 return " < " + typeName;
             }
+            else if (nameRegEx.IsMatch(this.Name.ToString()))
+            {
+                return " < " + "MsRestAzure::" + this.Name.ToString();
+            }
 
             return string.Empty;
         }
 
+        public static bool IsResourceModelsMatchStandardDefinition(CompositeType model)
```

Comment:
name of the method looks a bit weird, should it be "IsResourceModelMatchingStandardDefinition" or "resourceModelMatchStandardDefinition?" ?